tasks/run-collectors/run-collectors.yaml

```diff
@@ -0,0 +1,94 @@
+            exit 1
+        fi
+
+        git clone "${COLLECTORS_REPOSITORY}" collectors
+        pushd collectors
+
+        if [ ! -f "${RELEASEPLAN_FILE}" ] ; then
+            echo "No valid releasePlan file was found so no tenant collectors will be run."
+        else
+            NUM_TENANT_COLLECTORS="$(jq '.spec.collectors | length' "${RELEASEPLAN_FILE}")"
+            for ((i = 0; i < NUM_TENANT_COLLECTORS; i++)) ; do
+                collector="$(jq -c --argjson i "$i" '.spec.collectors[$i]' "${RELEASEPLAN_FILE}")"
+                name="$(jq -r '.name' <<< "$collector")"
+                type="$(jq -r '.type' <<< "$collector")"
+                TIMEOUT="$(jq -r --arg default "$DEFAULT_TIMEOUT" '.timeout // $default' <<< "$collector")"
+                ARG_STR=
+                NUM_ARGS="$(jq '.params | length' <<< "$collector")"
+                for ((j = 0; j < NUM_ARGS; j++)) ; do
+                    param="$(jq -c --argjson j "$j" '.params[$j]' <<< "$collector")"
+                    arg_name="$(jq -r '.name' <<< "$param")"
+                    arg_value="$(jq -r '.value' <<< "$param")"
+                    ARG_STR="${ARG_STR} --${arg_name} ${arg_value}"
+                done
+                # Execute collector
+                # Shellcheck wants ${ARG_STR} quoted, but that injects ' ' around the string, causing an error
+                # shellcheck disable=SC2086
+                timeout "$TIMEOUT" python3 "$type" ${ARG_STR} | tee "$RESULTS_DIR"/tenant-"$name".json
+            done
+        fi
+
+        if [ ! -f "${RELEASEPLANADMISSION_FILE}" ] ; then
+            echo "No valid releasePlanAdmission file was found so no managed collectors will be run."
+        else
+            NUM_MANAGED_COLLECTORS="$(jq '.spec.collectors | length' "${RELEASEPLANADMISSION_FILE}")"
+            for ((i = 0; i < NUM_MANAGED_COLLECTORS; i++)) ; do
+                collector="$(jq -c --argjson i "$i" '.spec.collectors[$i]' "${RELEASEPLANADMISSION_FILE}")"
+                name="$(jq -r '.name' <<< "$collector")"
+                type="$(jq -r '.type' <<< "$collector")"
+                TIMEOUT="$(jq -r --arg default "$DEFAULT_TIMEOUT" '.timeout // $default' <<< "$collector")"
+                ARG_STR=
+                NUM_ARGS="$(jq '.params | length' <<< "$collector")"
+                for ((j = 0; j < NUM_ARGS; j++)) ; do
+                    param="$(jq -c --argjson j "$j" '.params[$j]' <<< "$collector")"
+                    arg_name="$(jq -r '.name' <<< "$param")"
+                    arg_value="$(jq -r '.value' <<< "$param")"
+                    ARG_STR="${ARG_STR} --${arg_name} ${arg_value}"
+                done
+                # Execute collector
+                # Shellcheck wants ${ARG_STR} quoted, but that injects ' ' around the string, causing an error
+                # shellcheck disable=SC2086
+                timeout "$TIMEOUT" python3 "$type" ${ARG_STR} | tee "$RESULTS_DIR"/managed-"$name".json
```

Comment:
Isn't there a risk that two collectors will have the same name? Will we have a CI check in the git repo to prevent that?

Reply:
No, I don't plan to use any CI checks. There is no reason for a user to do this. If they do, they will get weird behavior, just like if a user specifies the same component name twice in their RPA mapping

Reply:
ok, sounds good